endpoints/setuid.go

```diff
@@ -135,16 +119,14 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 		tcf2Cfg := tcf2CfgBuilder(cfg.GDPR.TCF2, account.GDPR)
 
 		if shouldReturn, status, body := preventSyncsGDPR(gdprRequestInfo, gdprPermsBuilder, tcf2Cfg); shouldReturn {
-			w.WriteHeader(status)
-			w.Write([]byte(body))
```

Comment:
Why did we remove this line? In order for the refactored version to be completely equivalent, shouldn't we keep it?
```
121         if shouldReturn, status, body := preventSyncsGDPR(gdprRequestInfo, gdprPermsBuilder, tcf2Cfg); shouldReturn {
    +           w.Write([]byte(body))
122             var metricValue metrics.SetUidStatus
123             switch status {
124             case http.StatusBadRequest:
125                 metricValue = metrics.SetUidBadRequest
126             case http.StatusUnavailableForLegalReasons:
127                 metricValue = metrics.SetUidGDPRHostCookieBlocked
128             }
129             handleBadStatus(w, status, metricValue, errors.New(body), metricsEngine, &so)
130             return
131         }
endpoints/setuid.go
```

Reply:
So after looking into it more, this `body` value that `preventSyncsGDPR` is essentially a custom error message. So I don't think there's much of a difference between `w.Write([]byte(body))` and `w.Write([]byte(errors.New(body)) ` which is how it's written in `handleBadStatus`.

So I'd advocate for keeping the code how it is, and keeping it a bit cleaner. However, it is a very small difference, so I'm not totally against changing it either. What do you think?